lib/authority.js

```diff
@@ -214,12 +221,19 @@ Authority.prototype._validateViaInstanceDiscovery = function(callback) {
  * @param {Authority.GetOauthEndpointsCallback}  callback  The callback function.
  */
 Authority.prototype._getOAuthEndpoints = function(tenantDiscoveryEndpoint, callback) {
-  if (this._tokenEndpoint) {
+  if (this._tokenEndpoint && this._deviceCodeEndpoint) {
     callback();
     return;
   } else {
     // fallback to the well known token endpoint path.
-    this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    if (!this._tokenEndpoint){
+       this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    }
+
+    if (!this._deviceCodeEndpoint){
+       this._deviceCodeEndpoint = url.format(this._url) + AADConstants.DEVICE_ENDPOINT_PATH;
```

Comment:
I did test for that, and I'm not seeing url.format add a / at the end. Also from what's presented here https://nodejs.org/api/url.html, it doesn't look like it adds the slash at the end.